src/Debug/Stopwatch/Connection.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+declare(strict_types=1);
+
+namespace atk4\dsql\Debug\Stopwatch;
+
+use atk4\dsql\Connection_Proxy;
+use atk4\dsql\Expression;
+
+class Connection extends Connection_Proxy
+{
+    public $driverType = 'stopwatch';
+
+    /**
+     * Callable to call for outputting.
+     *
+     * Will receive parameters:
+     *  - Expression Expression object
+     *  - float      How long it took to execute expression
+     *  - boolean    True if we had exception while executing expression
+     *
+     * @var callable
+     */
+    public $callback;
+
+    /**
+     * @var float
+     */
+    protected $startTime;
+
+    /**
+     * Execute expression.
+     *
+     * @return \PDOStatement
+     */
+    public function execute(Expression $expr)
+    {
+        $this->startTime = microtime(true);
+
+        try {
+            $ret = parent::execute($expr);
+
+            $this->dump($expr);
+        } catch (\Exception $e) {
+            $this->dump($expr, true);
+
+            throw $e;
+        }
+
+        return $ret;
+    }
+
+    protected function dump(Expression $expr, $error = false)
+    {
+        $error = $error ? 'ERROR' : '';
```

Comment:
To be consistent with the other class `'ERROR '` (with space at the end) and no space below. Otherwise PR for Debug classes refactor complete.